outputs/azure.go

```diff
@@ -25,35 +25,45 @@ func NewEventHubClient(config *types.Configuration, stats *types.Statistics, pro
 
 // EventHubPost posts event to Azure Event Hub
 func (c *Client) EventHubPost(falcopayload types.FalcoPayload) {
-	log.Printf("[INFO]  : Try sending event")
+	c.Stats.AzureEventHub.Add(Total, 1)
+
+	log.Printf("[INFO] : %v EventHub - Try sending event", c.OutputType)
 	hub, err := eventhub.NewHubWithNamespaceNameAndEnvironment(c.Config.Azure.EventHub.Namespace, c.Config.Azure.EventHub.Name)
 	if err != nil {
-		c.Stats.AzureEventHub.Add(Error, 1)
-		c.PromStats.Outputs.With(map[string]string{"destination": "azureeventhub", "status": Error}).Inc()
+		c.setEventHubErrorMetrics()
 		log.Printf("[ERROR] : %v EventHub - %v\n", c.OutputType, err.Error())
 		return
 	}
+
 	log.Printf("[INFO]  : %v EventHub - Hub client created\n", c.OutputType)
-	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 
-	defer cancel()
 	data, err := json.Marshal(falcopayload)
 	if err != nil {
-		c.Stats.AzureEventHub.Add(Error, 1)
-		c.PromStats.Outputs.With(map[string]string{"destination": "azureeventhub", "status": Error}).Inc()
+		c.setEventHubErrorMetrics()
 		log.Printf("[ERROR] : Cannot marshal payload: %v", err.Error())
 		return
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
 	err = hub.Send(ctx, eventhub.NewEvent(data))
 	if err != nil {
-		c.Stats.AzureEventHub.Add(Error, 1)
-		c.PromStats.Outputs.With(map[string]string{"destination": "azureeventhub", "status": Error}).Inc()
+		c.setEventHubErrorMetrics()
 		log.Printf("[ERROR] : %v EventHub - %v\n", c.OutputType, err.Error())
 		return
 	}
+
+	// Setting the success status
+	go c.CountMetric(Outputs, 1, []string{"output:azureeventhub", "status:ok"})
 	c.Stats.AzureEventHub.Add(OK, 1)
 	c.PromStats.Outputs.With(map[string]string{"destination": "azureeventhub", "status": OK}).Inc()
-	log.Printf("[INFO]  : Successfully sent event")
+	log.Printf("[INFO] : %v EventHub - Publish OK", c.OutputType)
+}
 
-	c.Stats.AzureEventHub.Add(Total, 1)
+// setEventHubErrorMetrics set the error stats
+func (c *Client) setEventHubErrorMetrics() {
```

Comment:
I'm approving the MR, but I'm surprised you change the pattern for error stats for only this ouput